src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -825,6 +816,18 @@ fn configure_llvm(builder: &Builder<'_>, target: TargetSelection, cfg: &mut cmak
         }
     }
 
+    // Libraries for ELF section compression.
+    if !target.is_windows() {
+        cfg.define("LLVM_ENABLE_ZLIB", "ON");
+    } else {
+        cfg.define("LLVM_ENABLE_ZLIB", "OFF");
+    }
+
+    if builder.config.llvm_libzstd(target) {
+        cfg.define("LLVM_ENABLE_ZSTD", "FORCE_ON");
+        cfg.define("LLVM_USE_STATIC_ZSTD", "TRUE");
+    }
```

Comment:
Since the default value of `LLVM_ENABLE_ZSTD` is `ON`, if I understand its behavior correctly, it will pick up ZSTD *only* if it is available; if it is not available, it will not be included. Depending silently on the global environment in this way is not great, IMO. So I would prefer adding an `else` branch here that will set `LLVM_ENABLE_ZSTD` to `OFF`.

Reply:
Done.